contracts/Enigma.sol

```diff
@@ -170,16 +210,32 @@ contract Enigma is EnigmaStorage, EnigmaEvents, Getters, Ownable {
     * worker. This should be called by every worker (and the principal)
     * node in order to receive tasks.
     *
+    * @param _stakingAddress The operating address
     * @param _signer The signer address, derived from the enclave public key
     * @param _report The RLP encoded report returned by the IAS
     * @param _signature Signature
     */
-    function register(address _signer, bytes memory _report, bytes memory _signature)
+    function register(address _stakingAddress, address _signer, bytes memory _report, bytes memory _signature)
     public
     isUpdatedEnigmaContract
+    workerUnregistered(msg.sender)
```

Comment:
Why do you check that the worker is not registered? Isn't the `isUniqueSigningKey` require is enough?
If it is necessary, comparing only to the `Registered` status is not enough, other statuses should be checked as well...

Reply:
We felt it was necessary to add a `workerUnregistered` to make sure that the worker is currently unregistered in any capacity. `isUniqueSigningKey` only ensures that the signing address is unique, but there is no scenario that prevents a worker's ETH address from reregistering with a different signing address. This check prevents that from happening. Let me know what you think!

Reply:
So perhaps it will be better to call the require `isUniqueOperatingAddress`, it will be more clear (to me, at least) ..
In any case the check that the `state != Registered` in not enough, because it can be either `LoggedIn` or `LoggedOut` and the check will pass, although it should not

Reply:
agreed on renaming, i can make that edit!
as for the other point, let me know if you feel im looking at the wrong thing...but `workerUnregistered` is defined as `require(worker.status == EnigmaCommon.WorkerStatus.Unregistered, "Registered worker");` not `!= Registered` (which if that were the case, i would agree would be problematic). so i think it is implemented properly, but feel free to point me to where you see otherwise :)

Reply:
Ohh, my mistake, you are correct, the implementation of the `workerUnregistered` require is correct!